File: Selenium/Activities/Activity17.java
package activities;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.By;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.support.ui.Select;

import java.util.List;

public class Activity17 {
    public static void main(String[] args){
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        driver.get("https://v1.training-support.net/selenium/selects");
        System.out.println("Title is :" + driver.getTitle());
        WebElement select_element = driver.findElement(By.id("single-select"));
        Select select = new Select(select_element);
        select.selectByVisibleText("Option 2");
        System.out.println("select by Visible Text is :" + select.getFirstSelectedOption().getText());
        select.selectByIndex(3);
        System.out.println("select by Index is :" + select.getFirstSelectedOption().getText());
        select.selectByValue("4");
        System.out.println("select by Value is :" + select.getFirstSelectedOption().getText());
        List<WebElement> Options_List = select.getOptions();
        //System.out.println("select by Visible Text is :" + select.getAllSelectedOptions().toString());
        for (WebElement list : Options_List){
            System.out.println(list.getText());
        }
        driver.quit();
    }
}

File: Selenium/Activities/Activity17.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.ui import Select
from webdriver_manager.firefox import GeckoDriverManager

service = FirefoxService(GeckoDriverManager().install())
with webdriver.Firefox(service=service) as driver:
    driver.get("https://v1.training-support.net/selenium/selects")
    print("Title is :", driver.title)
    select_element = driver.find_element(By.ID,"single-select")
    dropdown = Select(select_element)
    dropdown.select_by_visible_text("Option 2")
    print("Select by Visible text Option is :", dropdown.first_selected_option.text)
    dropdown.select_by_index(3)
    print("Select by index Option is :", dropdown.first_selected_option.text)
    dropdown.select_by_value("4")
    print("Select by Value Option is :", dropdown.first_selected_option.text)
    list_elements = dropdown.options
    for lst in list_elements:
        print(lst.text)
    driver.quit()
